### PR TITLE
Search control component

### PR DESCRIPTION
**Pull request by kuzkok**

### CODE REVIEW

File: addon/components/base-layer.js [hunk 1]
```diff
@@ -139,20 +139,23 @@ export default Ember.Component.extend(
...
-      @param {Object[]} layer Object describing layer that must be searched.
+      @param {Object[]} layerModel Object describing layer that must be searched.
       @param {Object} searchOptions Search options related to layer type.
       @param {Object} results Hash containing search results.
       @param {Object[]} results.features Array containing (GeoJSON feature-objects)[http://geojson.org/geojson-spec.html#feature-objects]
       or a promise returning such array.
     */
     _search(e) {
-      let shouldSearch = e.layer === this.get('layer');
+      let shouldSearch = typeof (e.filter) === 'function' && e.filter(this);
```
Comment (SeregaMatin) [inline addon/components/base-layer.js]: Send outside a reference to a component instance (in e.filter callback), is bad practice.
Will be better to send JSON-object containing only filter-needed component's properties.

Also if search logic is refactored, then identification logic must be refactored  like that too.

File: addon/components/feature-result-item.js [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+/**
+  Component for display GeoJSON feature object details
+
+  @class FeatureResultItemComponent
+  @extends <a href="http://emberjs.com/api/classes/Ember.Component.html">Ember.Component</a>
+ */
+export default Ember.Component.extend({
+  /**
+    Flag: indicates where display detailed feature info
```
Comment (SeregaMatin) [inline addon/components/feature-result-item.js]: Maybe you mean "Flag: indicates whether to display detailed feature info or not".

File: addon/components/layer-result-list.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+/**
+  Component for display array of search\identify results
+
+  @class LayerResultListComponent
+  @extends <a href="http://emberjs.com/api/classes/Ember.Component.html">Ember.Component</a>
+ */
+export default Ember.Component.extend({
+
+  /**
+    Flag: indicates when one or more results contains more than 0 features
```
Comment (SeregaMatin) [inline addon/components/layer-result-list.js]: Maybe you mean "whether" instead of "when".

File: addon/components/layer-result-list.js [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+    @property _hasData
+    @type boolean
+    @default false
+   */
+  _hasData: false,
+
+  /**
+    Flag: inciates when one or more results.features promises rejected
```
Comment (SeregaMatin) [inline addon/components/layer-result-list.js]: Maybe you mean "whether" instead of "when".

File: addon/components/base-layer.js (hunk 1 above)
Comment (kuzkok) [inline addon/components/base-layer.js]: 1. fixed
2. identification logic be done later

**Pull request merged**